zio-http/src/test/scala/zhttp/service/ClientSpec.scala

```diff
@@ -1,16 +1,62 @@
 package zhttp.service
 
-import zhttp.internal.HttpRunnableSpec
-import zhttp.service.client.ClientSSLHandler.ClientSSLOptions
-import zio.test.Assertion.anything
-import zio.test.assertM
+import zhttp.http._
+import zhttp.internal.{AppCollection, HttpRunnableSpec}
+import zhttp.service.server._
+import zio.Task
+import zio.duration.durationInt
+import zio.test.Assertion._
+import zio.test.TestAspect._
+import zio.test._
 
 object ClientSpec extends HttpRunnableSpec(8082) {
-  val env           = ChannelFactory.auto ++ EventLoopGroup.auto()
-  override def spec = suite("Client")(
+
+  private val ClientUserAgentValue = "ZIO HTTP Client"
+
+  override def spec = {
+    suiteM("Client") {
+      serve(AppCollection.app).as(List(clientSpec)).useNow
+    }.provideCustomLayerShared(env) @@ timeout(5 seconds) @@ sequential
+  }
+  def clientSpec    = suite("ClientSpec") {
     testM("respond Ok") {
-      val actual = Client.request("http://api.github.com/users/zio/repos", ClientSSLOptions.DefaultSSL)
-      assertM(actual)(anything)
-    },
-  ).provideCustomLayer(env)
+      val app = Http.ok.requestBodyAsString()
+      assertM(app)(anything)
+    } +
+      testM("non empty content") {
+        val app             = Http.text("abc")
+        val responseContent = app.requestBody()
+        assertM(responseContent)(isNonEmpty)
+      } +
+      testM(
+        "POST request expect non empty response content from a server that requires a mandatory user agent header.",
+      ) {
+        val app = Http.text("test data") ++ Http.collectM[Request] { case req =>
+          val userAgentHeader       = req.getHeaderValue(Headers.Literals.Name.UserAgent)
+          val isExpectedHeaderValue = userAgentHeader.contains(ClientUserAgentValue)
+          if (isExpectedHeaderValue) req.getBodyAsString.map(text => Response.text(text))
+          else
+            Task.succeed(Response.status(Status.FORBIDDEN))
+
+        }
+
+        val userAgentHeader = Headers.userAgent(ClientUserAgentValue)
```

Comment:
If use agent is required as per the spec, then it should be handled automatically within the client. We should still have a test for it.

Reply:
@tusharmath in the http spec https://httpwg.org/specs/rfc7231.html#header.user-agent, the user agent SHOULD be set in request header, however some REST APIs are rejecting a client request that is not providing a valid User Agent header. One of those APIs is GitHub API: https://docs.github.com/en/rest/overview/resources-in-the-rest-api#user-agent-required.
I think that it is a good practice to provide user agent request header as part of the client HTTP request, so we might just to add  a default one.

Reply:
Got it. I think it would be nice to put a comment in the code about this requirement. `Client.scala` would be the best place.

**Side-Note:** I think the scope of this PR is changing quite a lot :) , I would rather just have the handler code merged and have the rest of the issues solved in a different PR.